websocket.go

```diff
@@ -609,8 +620,63 @@ func (c *wsConn) tryReconnect(ctx context.Context) bool {
 	return true
 }
 
+func (c *wsConn) readFrame(ctx context.Context, r io.Reader) {
+	// debug util - dump all messages to stderr
+	// r = io.TeeReader(r, os.Stderr)
+
+	// json.NewDecoder(r).Decode would read the whole frame as well, so might as well do it
+	// with ReadAll which should be much faster
+	// use a autoResetReader in case the read takes a long time
+	buf, err := io.ReadAll(c.autoResetReader(r)) // todo buffer pool
+	if err != nil {
+		c.readError <- xerrors.Errorf("reading frame into a buffer: %w", err)
+		return
+	}
+
+	c.frameExecQueue <- buf
+	if len(c.frameExecQueue) > cap(c.frameExecQueue)/2 {
+		log.Warnw("frame executor queue is backlogged", "queued", len(c.frameExecQueue), "cap", cap(c.frameExecQueue))
+	}
+
+	// got the whole frame, can start reading the next one in background
+	go c.nextMessage()
```

Comment:
The fact that we are starting goroutines only to kill them and start again bothers me a bit but it shouldn't matter apart from making goroutine numbers annoying to deal with.